data_model/src/permission.rs

```diff
@@ -1,212 +1,124 @@
 //! Permission Token and related impls
 #[cfg(not(feature = "std"))]
-use alloc::{borrow::ToOwned as _, collections::BTreeSet, format, string::String, vec::Vec};
+use alloc::{collections::BTreeSet, format, string::String, vec::Vec};
+use core::borrow::Borrow;
 #[cfg(feature = "std")]
 use std::collections::BTreeSet;
 
-use iroha_data_model_derive::model;
+use iroha_data_model_derive::{model, IdEqOrdHash};
 use iroha_schema::IntoSchema;
 use parity_scale_codec::{Decode, Encode};
 use serde::{Deserialize, Serialize};
 
 pub use self::model::*;
-use crate::name::Name;
+use crate::{executor::ExecutorDataModelObject, name::Name};
 
 /// Collection of [`Token`]s
 pub type Permissions = BTreeSet<PermissionToken>;
 
 use super::*;
 
-/// Unique id of [`PermissionToken`]
-pub type PermissionTokenId = Name;
-
 #[model]
 mod model {
     use super::*;
 
-    /// Stored proof of the account having a permission for a certain action.
-    ///
-    /// Since permission token is represented opaque to core
-    /// either executor or client should make sure that tokens are represented uniformly.
-    ///
-    /// So that:
-    /// - payload A is equal to payload B then token A must be equal to token B
-    /// - and if payload A is't equal to B then token A mustn't be equal to token B
+    /// Identification of [`PermissionToken`].
     #[derive(
         Debug,
+        Display,
         Clone,
         PartialEq,
         Eq,
         PartialOrd,
         Ord,
+        Hash,
+        Constructor,
+        FromStr,
+        Getters,
         Decode,
         Encode,
         Deserialize,
         Serialize,
         IntoSchema,
     )]
-    #[ffi_type]
-    pub struct PermissionToken {
-        /// Token identifier
-        pub definition_id: PermissionTokenId,
-        /// JSON encoded token payload
-        pub payload: JsonString,
+    #[getset(get = "pub")]
+    #[serde(transparent)]
+    #[repr(transparent)]
+    #[ffi_type(opaque)]
+    pub struct PermissionTokenId {
```

Comment:
```suggestion
    pub struct PermissionId {
```

I think we can rename `PermissionToken` to just `Permission`

Reply:
Done. Bloated the overall diff significantly...